File: 11-container-with-most-water/11-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& arr) {
        int i=0,j=arr.size()-1;
        int max_area=min(arr[i],arr[j])*(j-i);
        while(i<j){
            int current_area;
            if(arr[i]<arr[j]){
                    i++;
                current_area=min(arr[i],arr[j])*(j-i);
            }
            else{
                    j--;
               current_area=min(arr[i],arr[j])*(j-i);
            }
            max_area=max(current_area,max_area);
        }
        return max_area;
    }
};

File: 11-container-with-most-water/11-container-with-most-water.php
<?php

    class Solution {

    /**
     * @param Integer[] $height
     * @return Integer
     */
    function maxArea($height) {
        $left = 0;
        $right = count($height) - 1;
        $maxArea = 0;
        while ($left < $right) {
            $maxArea = max($maxArea, min($height[$left], $height[$right]) * ($right - $left));
            if ($height[$left] < $height[$right]) {
                $left++;
            } else {
                $right--;
            }
        }
        return $maxArea;
    }
}

?>